pyflakes/test/test_other.py

```diff
@@ -13,6 +13,97 @@ class Test(TestCase):
     def test_duplicateArgs(self):
         self.flakes('def fu(bar, bar): pass', m.DuplicateArgument)
 
+    def test_duplicate_keys(self):
+        self.flakes("{'yes': 1, 'yes': 1}", m.DuplicateDictionaryKey)
+
+    def test_multiple_duplicate_keys(self):
+        self.flakes(
+            "{'yes': 1, 'yes': 1, 'no': 2, 'no': 3}",
+            m.DuplicateDictionaryKey, m.DuplicateDictionaryKey)
+
+    def test_duplicate_keys_in_function(self):
+        self.flakes('''
+        def f(thing):
+            pass
+        f({'yes': 1, 'yes': 4})
+        ''', m.DuplicateDictionaryKey)
+
+    def test_duplicate_keys_in_lambda(self):
+        self.flakes("lambda x: {(0,1): 1, (0,1): 1}",
+                    m.DuplicateDictionaryKey)
+
+    def test_duplicate_keys_tuples(self):
+        self.flakes("{(0,1): 1, (0,1): 1}", m.DuplicateDictionaryKey)
+
+    def test_duplicate_keys_ints(self):
+        self.flakes("{1: 1, 1: 1}", m.DuplicateDictionaryKey)
+
+    def test_duplicate_keys_bools(self):
+        if version_info < (3, 4):
+            expected = m.DuplicateVariableDictionaryKey
+        else:
+            expected = m.DuplicateDictionaryKey
+        self.flakes("{True: 1, True: 1}", expected)
```

Comment:
IMO pyflakes should hide the fact that the ast is slightly different for these constants (True, False, None) for 3.4+ , so that the error message can be consistent between Python versions.  Our users dont consider `True` to be a variable on Python < 3.4.